sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/_settings_client.py

```diff
@@ -61,17 +63,18 @@ def extract_data(_):
         return ItemPaged(get_next, extract_data)
 
     @distributed_trace
-    def update_setting(self, name: str, value: str, **kwargs) -> KeyVaultSetting:
+    def update_setting(self, name: str, value: Union[bool, str], **kwargs) -> KeyVaultSetting:
```

Comment:
I'm wondering if there would be a benefit to overloads here:
```
def update_setting(self, name: str, value: bool, **kwargs) -> KeyVaultSetting[bool]:
    ...
def update_setting(self, name: str, value: str, **kwargs) -> KeyVaultSetting[str]:
    ...
```
However this would rely on a pretty strict distinction - which I don't think this API currently gives us. For example, I think I would pass in the value as `"true"` and this would actually update a boolean property, not a string property right?

Reply:
If we can use an overload to specify each `value` type but keep a plain `KeyVaultSetting` return type, that sounds like it could still be consistent. But yeah, sending a string value for a boolean setting would return a setting with a boolean `value`